clang/lib/Sema/SemaExpr.cpp

```diff
@@ -3907,6 +3907,24 @@ ExprResult Sema::ActOnNumericConstant(const Token &Tok, Scope *UDLScope) {
 
     Res = FixedPointLiteral::CreateFromRawInt(Context, Val, Ty,
                                               Tok.getLocation(), scale);
+  } else if (Literal.isDecimalFloatLiteral()) {
+    QualType Ty;
+
+    if (Literal.isDecimal32)
+      Ty = Context.DecimalFloat32Ty;
+    else if (Literal.isDecimal64)
+      Ty = Context.DecimalFloat64Ty;
+    else if (Literal.isDecimal128)
+      Ty = Context.DecimalFloat128Ty;
+
+    unsigned bit_width = Context.getTypeInfo(Ty).Width;
+
+    llvm::APDecimalFloat Val(llvm::decFltSemantics(
+        32u, /*Scale=*/1u, /*IsSigned=*/0u, /*IsSaturated=*/false,
+        /*HasUnsignedPadding=*/false));
```

Comment:
This will presumably need to be generalized for 64-bit and 128-bit literal support.